src/controller/python/chip/ChipStack.py

```diff
@@ -318,6 +318,7 @@ def setLogFunct(self, logFunct):
             self._ChipStackLib.pychip_Stack_SetLogFunct(logFunct)
 
     def Shutdown(self):
+        self._persistentStorage = None
```

Comment:
So, what exactly happens if this line is not added?

Reply:
We later fail to access the builtins.chipStack:
AttributeError: module 'builtins' has no attribute 'chipStack'

Reply:
From what I understand, `PersistentStorage.__del__` is called much later, presumably after `builtins.chipStack` has been cleaned up?

```
2022-05-13 18:48:30 allenwind chip.DL[974255] ERROR System Layer shutdown                                                                                                                                                                                                                                                     
[Thread 0x7ffff57d0640 (LWP 974259) exited]                                                                                                                                                                                                                                                                                   
Exception ignored in: <function PersistentStorage.__del__ at 0x7ffff4b8fca0>                                                                                                                                                                                                                                                  
Traceback (most recent call last):                                                                                                                                                                                                                                                                                            
  File "/home/sag/projects/project-chip/connectedhomeip/out/python_env/lib/python3.9/site-packages/chip/storage/__init__.py", line 157, in __del__
AttributeError: module 'builtins' has no attribute 'chipStack'
```

Reply:
By setting it to None, is this then re-ordering the destruction order such that PersistentStorage is getting deleted first before ChipStack?

Reply:
> By setting it to None, is this then re-ordering the destruction order such that PersistentStorage is getting deleted first before ChipStack?

Yes, that is exactly what is happening, before this change:

```
In [1]: quit
[Thread 0x7fffdf7fe640 (LWP 1452570) exited]
2022-05-18 15:51:03 allenwind chip.BLE[1452557] ERROR BleConnectionDelegate::CancelConnection is not implemented.
2022-05-18 15:51:03 allenwind chip.DL[1452557] ERROR renamed tmp file to file (/tmp/chip_counters.ini)
2022-05-18 15:51:03 allenwind chip.DL[1452557] ERROR Inet Layer shutdown
2022-05-18 15:51:03 allenwind chip.DL[1452557] ERROR BLE shutdown
2022-05-18 15:51:03 allenwind chip.DL[1452557] ERROR System Layer shutdown
[Thread 0x7ffff57cf640 (LWP 1452562) exited]
Destroying storage
Exception ignored in: <function PersistentStorage.__del__ at 0x7ffff4aefe50>
Traceback (most recent call last):
  File "/home/sag/projects/project-chip/connectedhomeip/out/python_env/lib/python3.9/site-packages/chip/storage/__init__.py", line 159, in __del__
AttributeError: module 'builtins' has no attribute 'chipStack'
```

After this change:

```
In [1]: quit
[Thread 0x7fffdf7fe640 (LWP 1453211) exited]
Destroying storage
2022-05-18 15:52:31 allenwind chip.BLE[1453201] ERROR BleConnectionDelegate::CancelConnection is not implemented.
2022-05-18 15:52:31 allenwind chip.DL[1453201] ERROR renamed tmp file to file (/tmp/chip_counters.ini)
2022-05-18 15:52:31 allenwind chip.DL[1453201] ERROR Inet Layer shutdown
2022-05-18 15:52:31 allenwind chip.DL[1453201] ERROR BLE shutdown
2022-05-18 15:52:31 allenwind chip.DL[1453201] ERROR System Layer shutdown
[Thread 0x7ffff57cf640 (LWP 1453204) exited]
[Thread 0x7fffdffff640 (LWP 1453210) exited]
[Thread 0x7fffecb3a640 (LWP 1453209) exited]
[Thread 0x7fffed33b640 (LWP 1453208) exited]
[Thread 0x7fffedb3c640 (LWP 1453207) exited]
[Thread 0x7fffee33d640 (LWP 1453206) exited]
[Inferior 1 (process 1453201) exited normally]
```


I'll add a comment.